src/main/scala/org/hammerlab/magic/test/listener/TestSparkListener.scala

```diff
@@ -158,7 +158,12 @@ object TestSparkListener {
 
   var instance: TestSparkListener = _
 
-  def apply(): TestSparkListener = instance
+  def apply(): TestSparkListener = {
+    if (instance == null) {
+      instance = new TestSparkListener()
```

Comment:
Actually it would still work, because when application is created we reset instance anyway, and subsequent calls just use that instance, but I will try to come up with better approach.